src/main/java/decodes/tsdb/CpCompDependsUpdater.java

```diff
@@ -620,15 +614,12 @@ private boolean tsCreated(DbKey tsKey)
 		{
 			String msg = "Error processing TS_CREATED for key=" + tsKey + ": " + ex;
 			warning(msg);
-			System.err.println(msg);
-			ex.printStackTrace(System.err);
+			StringWriter sw = new StringWriter();
+			PrintWriter pw = new PrintWriter(sw);
+			ex.printStackTrace(pw);
+			warning(sw.toString());
```

Comment:
sw never initialized to anything here?

Reply:
`StringWriter sw = new StringWriter()` is the initialization. StringWriter is a "StringBuffer" but for a print stream.

Reply:
right, but does printStackTrace(pw) cause sw to get a value as well? Sorry, don't know these interfaces well.

Reply:
no, the PrintWriter is initalized with the StringWriter, which initialized to the empty string.
StringWriter is the low level interface, PrintWriter provides the same interface as System.out basically. So `::printStackTrace(pw)` will write to the StringWriter, through the PrinterWriter, as though it was just outputing to stdout.

As to way Java decided they needed that many levels of indirection for this particular case; you'd have to ask the language/library designers.